planemo/shed_lint.py

```diff
@@ -311,7 +311,9 @@ def _lint_if_present(key, func, *args):
         if value is not None:
             msg = func(value, *args)
             if msg:
-                lint_ctx.warn(msg)
+                lint_ctx.error(msg)
+        else:
+            lint_ctx.error("Repository does not define: %s" % key)
```

Comment:
I think this is going against the original intent of the method `_lint_if_present` ? E.g. `type` is `unrestricted` by default, so probably no need to give an error if it's not present.

Reply:
Maybe we can rename the function? My aim it to error in cases where `shed_lint` will error ... and I guess this is the case here.

Reply:
OK. I see .. the type does not need to be defined in `.shed.yml`.

Reply:
Is this also the case for `name`, `owner`, and `categories`?

Reply:
`name` and `owner` can be specified on the command-line with e.g. `shed_update`. `name` defaults to the directory name if it's not specified in other ways. Categories I think are optional? I suppose these can all be warnings if not present.

Reply:
I think `name` and `owner` should also be in the realized repo if given on the command line: https://github.com/galaxyproject/planemo/blob/157cace58d4761cb42a8003d47221e55dce95da9/planemo/shed/__init__.py#L1022 (but I do not understand the code for setting `name` ..)